srml/support/src/storage/storage_items.rs

```diff
@@ -50,109 +50,129 @@ pub use crate::rstd::marker::PhantomData;
 #[doc(hidden)]
 pub use crate::rstd::boxed::Box;
 
+#[doc(hidden)]
+pub fn id<T>(t: T) -> T {
+	t
+}
+
+#[doc(hidden)]
+pub fn unwrap_or_default<T: Default>(t: Option<T>) -> T {
+	t.unwrap_or_else(|| Default::default())
+}
+
+#[doc(hidden)]
+pub fn wrap<T>(t: T) -> Option<T> {
+	Some(t)
+}
+
+#[doc(hidden)]
+pub fn require<T: Default>(t: Option<T>) -> T {
+	t.expect("Required values must be in storage")
+}
+
 // FIXME #1466 Remove this in favor of `decl_storage` macro.
 /// Declares strongly-typed wrappers around codec-compatible types in storage.
 #[macro_export]
 macro_rules! storage_items {
 	// simple values
 	($name:ident : $key:expr => $ty:ty; $($t:tt)*) => {
-		$crate::__storage_items_internal!(() () (OPTION_TYPE Option<$ty>) (get) (take) $name: $key => $ty);
+		$crate::__storage_items_internal!(() () (OPTION_TYPE Option<$ty>) (id) (id) $name: $key => $ty);
 		storage_items!($($t)*);
 	};
 	(pub $name:ident : $key:expr => $ty:ty; $($t:tt)*) => {
-		$crate::__storage_items_internal!((pub) () (OPTION_TYPE Option<$ty>) (get) (take) $name: $key => $ty);
+		$crate::__storage_items_internal!((pub) () (OPTION_TYPE Option<$ty>) (id) (id) $name: $key => $ty);
 		storage_items!($($t)*);
 	};
 	($name:ident : $key:expr => default $ty:ty; $($t:tt)*) => {
-		$crate::__storage_items_internal!(() () (RAW_TYPE $ty) (get_or_default) (take_or_default) $name: $key => $ty);
+		$crate::__storage_items_internal!(() () (RAW_TYPE $ty) (unwrap_or_default) (wrap) $name: $key => $ty);
 		storage_items!($($t)*);
 	};
 	(pub $name:ident : $key:expr => default $ty:ty; $($t:tt)*) => {
-		$crate::__storage_items_internal!((pub) () (RAW_TYPE $ty) (get_or_default) (take_or_default) $name: $key => $ty);
+		$crate::__storage_items_internal!((pub) () (RAW_TYPE $ty) (unwrap_or_default) (wrap) $name: $key => $ty);
```

Comment:
Can you not just directly pass `Some` here instead of `wrap`?

Reply:
Actually we can probably do that with this import on latest rust stable
```rust
#[doc(hidden)]
use Option::Some;
```